tests/unit/providers/agents/test_agent_metrics.py

```diff
@@ -0,0 +1,401 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
```

Comment:
these unit tests are extremely poor in taste (not your fault, but Sonnet's) because they use way too many Mocks. as such, they are incredibly fragile. please review them, and ask the AI to reconsider and rewrite.